### ISSUE TITLE
[kube-dns] runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by fhemberger** [kubernetes/kubeadm#934]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
```
&version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:00:59Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:13:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-12T14:14:26Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

- **Cloud provider or hardware configuration**:
Local machine

- **OS** (e.g. from /etc/os-release):
```
Ubuntu 16.04.4 LTS
```

- **Kernel** (e.g. `uname -a`):
```
Linux atom 4.4.0-119-generic #143-Ubuntu SMP Mon Apr 2 16:08:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

## What happened?
```
[dryrun] Resource name: "kube-dns"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x108 pc=0x2cb88ca]

goroutine 1 [running]:
k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns.translateStubDomainOfKubeDNSToProxyCoreDNS(0x3ace59a, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns/dns.go:281 +0x3a
k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns.coreDNSAddon(0xc420222000, 0x5979b00, 0xc420b16dc0, 0xc42111b400, 0xc4209048a8, 0xc420332310)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns/dns.go:156 +0x28a
k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns.EnsureDNSAddon(0xc420222000, 0x5979b00, 0xc420b16dc0, 0x0, 0x0)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns/dns.go:58 +0x1b9
k8s.io/kubernetes/cmd/kubeadm/app/cmd.(*Init).Run(0xc42049e010, 0x593c4e0, 0xc42000c018, 0x0, 0xc420510100)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/init.go:441 +0x1581
k8s.io/kubernetes/cmd/kubeadm/app/cmd.NewCmdInit.func1(0xc4200c7180, 0xc4201a89c0, 0x0, 0x4)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/init.go:139 +0x290
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200c7180, 0xc4201a8900, 0x4, 0x4, 0xc4200c7180, 0xc4201a8900)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42009c000, 0xc4200c7680, 0xc4200c7400, 0xc4200c7180)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:843 +0x334
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42009c000, 0xc42000c010, 0x593c4e0)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:791 +0x2b
k8s.io/kubernetes/cmd/kubeadm/app.Run(0xc420a3ff70, 0x4046f4)
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:36 +0xe3
main.main()
  /workspace/anago-v1.10.4-beta.0.68+5ca598b4ba5abb/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:27 +0x26
```

## What you expected to happen?
kube-dns to be setup successfully

## How to reproduce it (as minimally and precisely as possible)?
```
kubeadm init \
  --feature-gates CoreDNS=true \
  --dry-run --ignore-preflight-errors=all
```

**Comment by timothysc** [comment]
Yes, I believe this issue is fixed in 1.11 which will be coming in ~week+

**Comment by fhemberger** [comment]
@timothysc Woohoo, great news. Thanks!
